nilearn/maskers/tests/test_surface_maps_masker.py

```diff
@@ -0,0 +1,263 @@
+import numpy as np
+import pytest
+
+from nilearn._utils.class_inspect import check_estimator
+from nilearn.maskers import SurfaceMapsMasker
+from nilearn.surface import SurfaceImage
+
+
+@pytest.fixture
+def surf_maps_img(surf_mesh, rng):
+    """Return a sample surface map image using the sample mesh.
+    Has 6 regions in total: 3 in both, 1 only in left and 2 only in right.
+    Later we multiply the data with random "probability" values to make it
+    more realistic.
+    """
+    data = {
+        "left": np.asarray(
+            [
+                [1, 1, 0, 1, 0, 0],
+                [0, 1, 1, 1, 0, 0],
+                [1, 0, 1, 1, 0, 0],
+                [1, 1, 1, 0, 0, 0],
+            ]
+        ),
+        "right": np.asarray(
+            [
+                [1, 0, 0, 0, 1, 1],
+                [1, 1, 0, 0, 1, 1],
+                [0, 1, 1, 0, 1, 1],
+                [1, 1, 1, 0, 0, 1],
+                [0, 0, 1, 0, 0, 1],
+            ]
+        ),
+    }
+    # multiply with random "probability" values
+    data = {part: data[part] * rng.random(data[part].shape) for part in data}
+    return SurfaceImage(surf_mesh(), data)
+
+
+# tests for scikit-learn compatibility
+extra_valid_checks = [
+    "check_no_attributes_set_in_init",
+    "check_parameters_default_constructible",
```

Comment:
It seems this test enforces that we cannot have any required parameters while instantiating estimators. 

Would be a problem with maskers e.g. Nifti/SurfaceLabelsMasker or Nifti/SurfaceMapsMasker which have `labels_img` and `maps_img` as required parameters.

* Related to #4538

Reply:
This would mean change in behavior and consequently deprecation cycles for all maskers?

Reply:
yup at the moment I think we want to see how to maximize following the sklearn recommendations without breaking any API

when we introduce new things though let's try to comply as much as possible, no?

Reply:
Indeed, that makes sense. 

But just for my understanding, we would rather provide `maps_img` during `.fit()`?

Reply:
And should it be None at instantiation or should it not exist at all in the variables?

Reply:
> we would rather provide maps_img during .fit()?
ah no I think that there are some cases where we can have our own rules: in this case, I would pass this at instantiation

Reply:
Yeah I think then we need to have at least a default value of `None` for `maps_img` and `labels_img`:

https://scikit-learn.org/stable/developers/develop.html#instantiation

<img width="780" alt="Screenshot 2024-12-09 at 2 22 05 PM" src="https://github.com/user-attachments/assets/bf913b85-d5e1-498a-9456-970733457287">